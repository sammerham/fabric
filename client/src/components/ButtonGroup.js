import { React, useContext } from 'react'
import {
  StyledButton,
  StyledDivButton,
} from './StyledComponents'
import MediaContext from '../mediaContext'
import MediaApi from '../api/mediaApi';



/*
 * ButtonGroup
 *
 *  This is the page where you rednder mainTitle and main media buttons.
 * State: None

 * props:
      None
  
 * App ---->> Header --->> GroupButton
 */


function ButtonGroup() {
  const { setData } = useContext(MediaContext);
  // const { getAllMatrixMedia, getAllMatrixrevolutionsMedia, getAllMatrixreloadedMedia } = MediaApi;
  const { getMedia } = MediaApi;
  


// fn handle click to get the media data
  async function fetchMediaData(str) {
    const MediaData = await getMedia(str);
    setData(data => MediaData.Search);
  }

  return (
    <div>
      <StyledDivButton>
        <StyledButton bg={'#58755b'} onClick={() => fetchMediaData('matrix')}>Matrix</StyledButton>
        <StyledButton bg={'#58755b'} onClick={() => fetchMediaData('matrix revolutions')}>Matrix Revolutions</StyledButton>
        <StyledButton bg={'#58755b'} onClick={() => fetchMediaData('matrix reloaded')}>Matrix Reloaded</StyledButton> 
      </StyledDivButton>
    </div>
  )
}

export default ButtonGroup
