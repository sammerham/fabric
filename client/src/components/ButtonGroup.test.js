import React from 'react'
import { render } from '@testing-library/react';
import ButtonGroup from './ButtonGroup';
import MediaContext from '../mediaContext';

const MOCK = () => console.log('hello')


// smoke test
it('renders without crashing', () => {
    <ButtonGroup />
});

// snapShot
it("matches snapshot", function () {
  const { container } = render(
    <MediaContext.Provider value={{ MOCK }}>
      <ButtonGroup />
    </MediaContext.Provider>
  );
  expect(container).toMatchSnapshot();
});