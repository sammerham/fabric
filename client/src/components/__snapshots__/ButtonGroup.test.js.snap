// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <h1
      class="sc-iCfMLu cKMWxr App-sameh"
    >
      Media Information
    </h1>
    <div
      class="sc-gKclnd hrMEC"
    >
      <button
        class="sc-furwcr grhNup"
      >
        Matrix
      </button>
      <button
        class="sc-furwcr grhNup"
      >
        Matrix Revolutions
      </button>
      <button
        class="sc-furwcr grhNup"
      >
        Matrix Reloaded
      </button>
    </div>
    <div>
      <form
        class="sc-kDTinF jSvsQF"
      >
        <input
          class="sc-pVTFL xFLhP"
          id="term"
          name="term"
          placeholder="Enter a search Term"
          type="text"
          value=""
        />
        <button
          class="sc-jrQzAO bGNTCT"
        >
          Search
        </button>
      </form>
    </div>
  </div>
</div>
`;
